### DISCUSSION TITLE
Mapping of observed data and simulation outputs in simulations and further chart improvements

### DISCUSSION THREAD

**Discussion by PavelBal** [Open-Systems-Pharmacology/OSPSuite.Core#1628] [Ideas]
## Problem description
- All time profiles for a PI solution are shown in one plot, which makes it hard to distinguish between different simulations
- No GOF plots for simulations outside of PI available
- "Predicted vs. Observed" plots missing fold deviation line, a common way to assess the goodness of fit/prediction

## Proposed solution
Following features are to be implemented and will be described in detail in the following text:
1.	Mapping of simulation outputs to observed data in simulations
2.	Separating of simulation outputs from observed data in the Chart Editor of the “Time Profile”, “Predicted vs. Observed”, “Residual vs. Time” plots (individual simulations and parameter identification (PI)
3.	De-/Selecting a simulation output de-/selects the mapped observed data (individual simulation and PI)
4.	Allow chart types “Predicted vs. Observed”, “Residual vs. Time” for individual simulations
5.	Calculation of error between mapped simulation results and observed data in individual simuilations
6.	Add fold deviation lines for “Predicted vs. Observed” plots

The following sections describe the features in detail.

### Mapping of simulation outputs to observed data in simulations
Individual and population simulations should get a new tab “Data” next to the tabs “Parameters” and “Reaction Diagram” (PK-Sim only). The tab cannot be closed.
 
![image](https://user-images.githubusercontent.com/25903040/166730634-6411fae5-5c9d-40cb-8f64-a7c2c4e5bdd9.png)

The tab lists all observed data that has been added to any of the Time Profile plots of the respective simulation (in PK-Sim, multiple Time Profile plots can be created per simulation, in MoBi only one Time Profile). The data are listed in similar way as in the “Data” tab of the parameter identification, with some differences:
-	The column “Observed Data” is the first column, the column “Outputs” is the second
-	“Observed Data” is NOT a drop-down menu
-	Preview of observed data is not shown
 
![image](https://user-images.githubusercontent.com/25903040/166730678-69777fe5-0aa5-4e42-9b2b-087e0aa10162.png)

For each observed data, a simulation output can be selected. “None” can also be selected. When adding observed data, the software should try to map the respective simulation output based on the meta data of the observed data (the same way it is done for parameter identification)

In PK-Sim, observed data is NOT SHOWN in the simulation explorer any more:

![image](https://user-images.githubusercontent.com/25903040/166730752-a38dbc55-4999-4e43-9a3b-6f8dc5a5f3f0.png)
 
Removing the data from the “Data” tab removes it from all plots of the respective simulation where it is used (disable for population simulations?). 

### Separating of simulation outputs from observed data in the Chart Editor It is proposed to split the „Chart editor“ between simulation outputs and observed data. 
 
![image](https://user-images.githubusercontent.com/25903040/166730854-91d289ca-b36d-4c86-82df-59dd82876697.png)

Selecting or de-selecting a simulation output automatically selects (or de-selects) the observed data linked to that simulation. This option can be turned on and off by (un-)checking the “Link data to simulations” checkbox. Valid for PI and simulation plots

### Allow chart types “Predicted vs. Observed”, “Residual vs. Time” for individual simulations
Plot types “Predicted vs. Observed” and “Residuals vs. Time” are only available for PI results. They should also be enabled for individual simulations.
“Predicted vs. Observed” is created based on the mapping of observed data to simulation outputs defined in the tab “Data”. Only mapped observed data should be included (i.e., data mapped to “None” is not included).
For the “Residuals vs. Time” plot, the residuals are calculated based on the “Scaling” selection and the “Weight” of the entry (as defined in the “Data” tab, residuals calculation should be the same as in the PI). Weight can only be assigned to the whole data set and not to single data points. In addition, the total error (as calculated for the final PI result) should be shown, e.g. below the plot.
Re-running the simulation updates all plots (and re-calculates the error).
When changing anything in the “Data”-tab, e.g.
-	Changing (adding/removing) mapping
-	Changing the scaling property
-	Changing the weight
-	Removing data
“Residuals vs. Time” should be marked with a red exclamation mark (as the one used in PK-Sim to signal that Time Profile is not up to da) to make the user aware that the plots are not up-to-date. A “refresh”-button should re-create the plot without the necessity to close and create a new one. Background – changing one of the listed properties in the “Data” tab can result in different residuals and total error.

### Add deviation lines to GOF plots
It is also proposed to add deviation lines to GOF plots. The user will be able to introduce a new deviation line by right clicking on the plot and selecting “Add deviation line”. This will open a “New Deviation Line” Dialog window, where the user will be able to specify the fold factor of the deviation line (eg 2-fold, 3-fold and so on). 
 
![image](https://user-images.githubusercontent.com/25903040/166731087-3502633b-ac71-4ed2-a93c-a799d73d9a94.png)

After confirming the fold input, corresponding deviation lines will be added to the plot.

![image](https://user-images.githubusercontent.com/25903040/166731165-a0683222-7fc3-4d7d-9c99-3f64bec40914.png)

The newly introduced deviation lines will appear in the “Curves and Axis Options” of the chart, like fe the “Dashes” currently do. They will be named after their fold value. There the user will be able to select or deselect the created deviation lines, making them visible or not in the chart.
 
![image](https://user-images.githubusercontent.com/25903040/166731239-492f8987-7cce-4ce8-9a40-2807d739be8c.png)

**Comment by PavelBal** [comment]
@msevestre @Yuri05 @StephanSchaller @georgeDaskalakis

**Comment by StephanSchaller** [comment]
Nice!

> The column “Observed Data” is the first column, the column “Outputs” is the second

Call it "Simulation Outputs"?

> For the “Residuals vs. Time” plot, the residuals are calculated based on the “Scaling” selection and the “Weight” of the entry (as defined in the “Data” tab, residuals calculation should be the same as in the PI). Weight can only be assigned to the whole data set and not to single data points. In addition, the total error (as calculated for the final PI result) should be shown, e.g. below the plot.

This might be of general interest, yes, however, of specific interest might be the PK-Property ratios (AUCR, CmaxR) for the curves. These should also be calculated!

**Comment by PavelBal** [comment]
@msevestre @Yuri05
